### ISSUE TITLE
Restructure `computer` and `server` folders.

### ISSUE THREAD

**Issue by soupglasses** [numtide/srvos#223]
**Is your feature request related to a problem? Please describe.**

Currently a lot of folders inside `./nixos` follows a set of "concepts" so to say, like hardware, mixins, modules, and roles. It seems out of place that the default configs for servers and computers are put into this top level, especially since they are both essentially single files of `default.nix`. 

**Describe the solution you'd like**

This is very much up in the air, and very much for discussion. But something among the lines of `./nixos/kind/server.nix` or `./nixos/kinds/computer/default.nix` could be a better structure here to follow the idea of having top level "concepts".

**Describe alternatives you've considered**

Continuing having the `./nixos/computer` and `./nixos/server` folders.

**Additional context**

**Comment by Mic92** [comment]
Changing this now would break the import paths. srvos.nixosModules.server is bascially the default import for all servers.

**Comment by zimbatm** [comment]
What is the `computer` one referring to? We have "server", "desktop" and "common".

I understand the tension but would rather keep the folder structure as shallow as possible. If we add more "kinds" of machines then it would make sense to create that folder.

**Comment by soupglasses** [comment]
Yeah sorry. When I stated `computer` I was thinking about desktop. Sorry for the confusion
